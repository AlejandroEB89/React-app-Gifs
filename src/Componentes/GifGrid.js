// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {

    const { data:img , loading} = useFetchGifs( category );

    
    return ( 
        <>
            <h3 className="animate__animated animate__fadeIn"> { category } </h3>
            { loading && <p data-testid="pLoad" className="animate__animated animate__flash"> Cargando...</p> }
            <div className="card-grid">
            

                { img.map( imag => ( 
                    <GifGridItem 
                    key={ imag.id }
                    {...imag} />   //el op spread esta enviando el resto de propiedades que tiene el img como props al GifgridItem
                ) ) 
                }
            </div>
        </>
        
     );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
 
export default GifGrid;