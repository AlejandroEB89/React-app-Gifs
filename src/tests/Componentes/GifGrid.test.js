import { render, screen } from '@testing-library/react';
import GifGrid from '../../Componentes/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGrid', () => {
    const category = 'newCategory';
    const componente = <GifGrid key= { category } category={ category }/>;

    test('Debe renderizarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [] ,
            loading: true
         });
        
        const { asFragment } = render( componente );   //renderizo el componente como un obj Fragment para el snapshot
        expect( asFragment() ).toMatchSnapshot();
    })
    // Con jest.mock(...el hook) utilizamos un mock de nuestro hook para simular que nos devuelve la data
    test('Debe mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id: 'AAAA',
            title: 'Cualquier imagen',
            url: 'https://localhost.laimagen.jpg'
        }]
        
        useFetchGifs.mockReturnValue({         //simulamos datos que nos devuelve el usefetchgifs
            data: gifs ,
            loading: false
        });


        const { asFragment } = render( componente );
        expect( asFragment() ).toMatchSnapshot();

     
    })
})