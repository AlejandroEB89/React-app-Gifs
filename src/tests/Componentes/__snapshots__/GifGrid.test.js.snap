// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<DocumentFragment>
  <h3
    class="animate__animated animate__fadeIn"
  >
     newCategory 
  </h3>
  <div
    class="card-grid"
  >
    <div
      class="card animate__animated animate__fadeIn"
    >
      <img
        alt="Cualquier imagen"
        src="https://localhost.laimagen.jpg"
      />
      <p>
         Cualquier imagen
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Pruebas en GifGrid Debe renderizarse correctamente 1`] = `
<DocumentFragment>
  <h3
    class="animate__animated animate__fadeIn"
  >
     newCategory 
  </h3>
  <p
    class="animate__animated animate__flash"
    data-testid="pLoad"
  >
     Cargando...
  </p>
  <div
    class="card-grid"
  />
</DocumentFragment>
`;
